sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1257,306 +1263,234 @@ __ESIMD_API std::enable_if_t<std::is_floating_point<T>::value, T> asin(T src0) {
 
 /* atan2_fast - a fast atan2 implementation */
 /* vector input */
-template <int N>
-__ESIMD_NS::simd<float, N> atan2_fast(__ESIMD_NS::simd<float, N> y,
-                                      __ESIMD_NS::simd<float, N> x);
+template <typename T, int N>
```

Comment:
Not that this is an interface change. Prior to this change atan2_fast always returned float. Is this intentional?
Also applies to other APIs. Please preserve the original return types, unless there is some reason not to.